nipype/interfaces/base/tests/test_specs.py

```diff
@@ -63,6 +81,36 @@ def test_TraitedSpec_dynamic():
         assign_a_again
 
 
+def test_DynamicTraitedSpec_tab_completion():
+    def extract_func(list_out):
+        return list_out[0]
+
+    # Define interface
+    func_interface = Function(input_names=["list_out"],
+                             output_names=["out_file","another_file"],
+                             function=extract_func)
+    # Define node
+    list_extract = Node(Function(
+        input_names=["list_out"],output_names=["out_file"],
+        function=extract_func), name="list_extract")
+
+    # Check __all__ for interface inputs
+    expected_input = sorted(list_extract.inputs.editable_traits())
+    assert(sorted(func_interface.inputs.__all__) == expected_input)
+
+    # Check __all__ for node inputs
+    assert(sorted(list_extract.inputs.__all__) == expected_input)
+
+    # Check __all__ for node outputs
+    expected_output = sorted(list_extract.outputs.editable_traits())
+    assert(sorted(list_extract.outputs.__all__) == expected_output)
+
+    # Add trait and retest
+    list_extract._interface._output_names.append('added_out_trait')
+    expected_output = sorted(['added_out_trait',*expected_output])
```

Comment:
Python 2.7, 3.4 don't like the `[*x]` syntax. How about making `expected_output` a set above, moving to set comparisons, and changing this to:

```Python
expected_output.add('added_out_trait')
```

Reply:
nice. Thanks for the tip. done